src/parser/shared/modules/resources/resourcetracker/ResourceTracker.ts

```diff
@@ -13,20 +34,22 @@ class ResourceTracker extends Analyzer {
     // spellResourceCost: SpellResourceCost,
   };
 
+  protected eventEmitter!: EventEmitter;
+
   current = 0;
-  resourceUpdates = [];
+  resourceUpdates: ResourceUpdate[] = [];
 
   // stores resource gained/spent/wasted by ability ID
-  buildersObj = {};
-  spendersObj = {};
+  buildersObj: {[index: number]: BuilderObj} = {};
+  spendersObj: {[index: number]: SpenderObj} = {};
 
   // TODO set this to the resource you wish to track constructor.. see the appropriate objects in game/RESOURCE_TYPES
-  resource;
+  resource!: Resource;
```

Comment:
you could make the class + resource abstract so that extending classes will get an error if they dont set it, but this is fine for now